stacks-signer/src/client/stacks_client.rs

```diff
@@ -207,6 +207,9 @@ impl StacksClient {
             estimated_len: Some(tx.tx_len()),
             transaction_payload: to_hex(&tx.payload.serialize_to_vec()),
         };
+        #[cfg(feature = "monitoring_prom")]
+        let timer =
+            crate::monitoring::new_rpc_call_timer(&self.fees_transaction_path(), &self.http_origin);
```

Comment:
I think it might be nicer to locate the `cfg()` switch in the `monitoring` crate. So, `new_rpc_call_timer()` is defined both with `monitoring_prom` and without (without `monitoring_prom` it would just be an empty struct that no-ops on `stop_and_record`). That way, we wouldn't need as many `cfg` flags throughout the codebase. This isn't a strong opinion, though, so if you think this approach is better, that's fine too.